cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -24,21 +24,21 @@ import scala.util.Random
 /**
  * A Traverser traverses the graph in a certain order of nodes.
  */
-trait Traverser extends Iterator[Node] { self =>
+trait Traverser[+V <: Node] extends Iterator[V] { self =>
   /**
    * Traverser only visits node Ids listed in some node's edges, and thus
    * we know that node must exist, safe to assume graph.getNodeById return non-None value
    */
-  protected def getExistingNodeById(graph: Graph, id: Int) = graph.getNodeById(id).get
+  protected def getExistingNodeById[V <: Node](graph: Graph[V], id: Int) = graph.getNodeById(id).get
```

Comment:
I think, it is breaking the type bound.
I'm getting this error `type arguments [W] do not conform to trait Graph's type parameter bounds [+V <: com.twitter.cassovary.graph.Node]`

Here is some rough code
```
scala> trait Node
defined trait Node

scala> trait Graph[+V <: Node]
defined trait Graph

scala> trait Traverser[+V <: Node] {
     |   protected def someFunction[W >: V](graph: Graph[W]) = println("hi")
     | }
<console>:10: error: type arguments [W] do not conform to trait Graph's type parameter bounds [+V <: Node]
         protected def someFunction[W >: V](graph: Graph[W]) = println("hi")
 
```